client/packages/common/src/utils/BarcodeScannerContext.tsx

```diff
@@ -73,11 +72,29 @@ export const parseResult = (content?: string): ScanResult => {
       expiryDate: expiry ? Formatter.naiveDate(expiry) : undefined,
       gtin,
     };
-  } catch {
+  } catch (e) {
+    console.error(`Error parsing barcode ${content}:`, e);
     return { content };
   }
 };
 
+export const parseResult = (content?: string): ScanResult => {
+  if (!content) return {};
+
+  const result: ScanResult = { content };
+  // a data matrix result from the Capacitor barcode scanner plugin
+  // is returned in sections, separated by \x1D and needs to be parsed
+  // in sections - otherwise the serial number includes the AI characters
```

Comment:
ah, yes. perhaps need to update the comment. It's a GS1 term. From the docs (linked above) for the parsing library : `application identifier ("AI")` is the term for the numbers (01, 17, 10) that are used to identify GTIN, expiry & batch

Reply:
> is returned in sections, separated by \x1D and needs to be parsed

This also happens with normal scans with Zebra printer, and was already being dealt with, do you mean `\x1D` was returned as a string rather then String.fromCharCode(29) ? In this case I think we should just adjust capacitor scan input by replacing `\x1D` with String.fromCharCode(29)

Reply:
I didn't notice that! asci 29 and `\x1d`.
Have tried just replacing and that doesn't work - in fact I can't see how the group separator char is being handled, with them in place the scan just doesn't parse.

Reply:
The problem is actually that the group by char is at the start and the parsing library is just working with the first group. Removing all of the `\x1d` chars for the android scanner fixes the problem 😄 

Have removed the tests too, since they are no longer helpful.